packages/react/src/tanstackrouter.ts

```diff
@@ -0,0 +1,71 @@
+import { WINDOW, startBrowserTracingNavigationSpan, startBrowserTracingPageLoadSpan } from "@sentry/browser";
+import {
+	SEMANTIC_ATTRIBUTE_SENTRY_OP,
+	SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+	SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+} from "@sentry/core";
+
+import { browserTracingIntegration as originalBrowserTracingIntegration } from "@sentry/react";
+import type { Client, Integration } from "@sentry/types";
+
+import type { TanstackRouter } from "./types";
+
+/**
+ * A custom browser tracing integration for Tanstack Router.
+ */
+export function tanstackRouterBrowserTracingIntegration(
+	router: TanstackRouter,
+	options: Parameters<typeof originalBrowserTracingIntegration>[0] = {}
+): Integration {
+	const browserTracingIntegrationInstance = originalBrowserTracingIntegration({
+		...options,
+		instrumentNavigation: false,
+		instrumentPageLoad: false,
+	});
+
+	const { instrumentPageLoad = true, instrumentNavigation = true } = options;
+
+	return {
+		...browserTracingIntegrationInstance,
+		afterAllSetup(client) {
+			const initPathName = WINDOW && WINDOW.location && WINDOW.location.pathname;
+			if (instrumentPageLoad && initPathName) {
+				startBrowserTracingPageLoadSpan(client, {
+					name: initPathName,
+					attributes: {
+						[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: "url",
+						[SEMANTIC_ATTRIBUTE_SENTRY_OP]: "pageload",
+						[SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: "auto.pageload.react.tanstack_router",
+					},
+				});
+			}
+
+			if (instrumentNavigation) {
+				tanstackRouterInstrumentNavigation(router, client);
+			}
+
+			browserTracingIntegrationInstance.afterAllSetup(client);
+		},
+	};
+}
+
+export function tanstackRouterInstrumentNavigation(router: TanstackRouter, client: Client): void {
+	router.history.subscribe(() => {
+		const state = router.state;
+		const matches = state.pendingMatches ?? state.matches;
```

Comment:
it seems like what we do internally is more like this:

```
const matches = [
  ...state.cachedMatches,
  ...(state.pendingMatches ?? []),
  ...state.matches,
]
```

see: https://github.com/TanStack/router/blob/bce8e71652acdfa66594fedaab4bc5f058ccaf6e/packages/react-router/src/RouterProvider.tsx#L113-L117

I did find multiple implementations of this with different spreading order, so I'm not sure which one would be correct 😅

Reply:
You'll want one of these depending on the situation:
- `state.pendingMatches || state.matches` - The user may be in the process of navigating to a new location, which is why you would want pendingMatches to represent that pending navigation
- `state.matches` - If you're only interested in where the user is **currently** and are not interested in any potential pending navigation destinations, then this should suffice.

I recommend the former. Do not use `state.cachedMatches` for anything outside of reporting back stuff that's cached.

Reply:
@tannerlinsley I did `const matches = state.pendingMatches ?? state.matches;`. Would that be worse than `state.pendingMatches || state.matches`?